### DISCUSSION TITLE
vim-like hjkl style navigation

### DISCUSSION THREAD

**Discussion by tony** [extrawurst/gitui#1114] [Q&A] [answered]
With [Customizable key bindings merged in #234](https://github.com/extrawurst/gitui/pull/234), it may be helpful for earlier  requests at #57 and #227 to have an example of what a gitui config with `vim`-like navigation would be like.

Is this doable? As of what tigui version?

If so, what would an example configuration / initialization look like?

**Comment by extrawurst** [comment, marked as answer]
I thinks its all documented here: https://github.com/extrawurst/gitui/blob/master/KEY_CONFIG.md

**Reply by tony** [reply]
Thank you!

It looks like [`vim_style_key_config.ron`](https://github.com/extrawurst/gitui/blob/70ce3f8/vim_style_key_config.ron) has this

It seems @yanganto's #234 PR included this and it's been being improved ever since.

Initial config in [`a95ffd7`](https://github.com/extrawurst/gitui/blob/a95ffd7bcc55bacf8925a8d637f2d906b30932c3/assets/vim_style_key_config.ron) ([full commit](https://github.com/extrawurst/gitui/commit/a95ffd7bcc55bacf8925a8d637f2d906b30932c3))

History: old location @ [`/assets/vim_style_key_config.ron`](https://github.com/extrawurst/gitui/commits/b0b37a84e478ada5f87674998b572c33e7b4c93d/assets/vim_style_key_config.ron), new location @ [`/vim_style_key_config.ron`](https://github.com/extrawurst/gitui/commits/70ce3f88d69e5f7fcdc03837f3115ee91943dbd1/vim_style_key_config.ron)

**Comment by JurajLieskovsky** [comment]
As a new user I was really disappointed with the [vim_style_key_config](https://github.com/extrawurst/gitui/blob/904885e001bddd3c2b36311d69debbc26c3623ca/vim_style_key_config.ron) on master. It does not include
```
    focus_right: Some(( code: Char('l'), modifiers: ( bits: 0,),)),
    focus_left: Some(( code: Char('h'), modifiers: ( bits: 0,),)),
    focus_above: Some(( code: Char('k'), modifiers: ( bits: 0,),)),
    focus_below: Some(( code: Char('j'), modifiers: ( bits: 0,),)),
```
So in the "status" tab I could not move me from the "changes" column to the "diff".

**Reply by JurajLieskovsky** [reply]
I just checked and I am indeed on the newest release (v0.22.1)

**Reply by yanganto** [reply]
> So in the "status" tab I could not move me from the "changes" column to the "diff".
It works with `l` to move `Unstaged Changes` to `Diff` in 0.21.0 from my end.  Let me check this on latest version later.
